docs/developer/best_practices/commit.rst

```diff
@@ -0,0 +1,357 @@
+---------------------
+
+How to split a changes into commits
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The golden rule to create good commits is to ensure that there is only one
+logical change per commit.
+
+This has several advantages amongst whose:
+
+- small commits are easier to review (a large pull request correctly divided
+  into logical commits is easier/faster to review than a medium-sized one with
+  less thought-out division)
+
+- logical commits are easier to revert (`e866b01f0553 <https://github.com/scality/metalk8s/commit/e866b01f05535925e80da20aca00417904422433>`_/`8208a170ac66 <https://github.com/scality/metalk8s/commit/8208a170ac66912ace018bcd00c058ad214d169b>`_)/cherry-pick
+  (`Pull request #1641 <https://github.com/scality/metalk8s/pull/1641>`_)
+
+- when looking for a regression (e.g. using ``git bisect``) it is easier to
+  find the root cause
+
+- make ``git log`` and ``git blame`` way more useful
+
+Examples
+^^^^^^^^
+
+Cosmetic changes
+::::::::::::::::
+
+Use a dedicated commit when you want to make cosmetic changes to the code
+(linting, whitespaces, alignment, renaming, …).
+
+Mixing cosmetics and functional changes is bad because the cosmetics (which
+tends to generate a lot of diff/noise) will obscure the important functional
+changes, making it harder to correctly determine whether the change is correct
+during the review.
+
+Example (Pull request `#1620 <https://github.com/scality/metalk8s/issues/1620>`_):
+
+- one commit for the cosmetic changes: `766f572e462c6933c8168a629ed4f479bb68a803 <https://github.com/scality/metalk8s/commit/766f572e462c6933c8168a629ed4f479bb68a803>`_
+
+- one commit for the functional changes: `3367fabdefc0b35d34bf7cf2fb0d33ff81f9fd5a <https://github.com/scality/metalk8s/commit/3367fabdefc0b35d34bf7cf2fb0d33ff81f9fd5a>`_
+
+Refactoring
+:::::::::::
+
+When introducing new features, you often have to add new helpers or refactor
+existing code. In such case, instead of having single commit with everything
+inside, you can either:
+
+1. first add a new helper: `29f49cbe9dfa <https://github.com/scality/metalk8s/commit/29f49cbe9dfa0b824c818d25d4a2f6965351e65d>`_
+
+2. then use it in new code: `7e47310a8f20 <https://github.com/scality/metalk8s/commit/7e47310a8f20fd49f0ad36707b20e6c2a53df638>`_
+
+Or:
+
+1. first add the new code: `5b2a6d5fa498 <https://github.com/scality/metalk8s/commit/5b2a6d5fa49815180a2effdd37cb58542e83b5a5>`_
+
+2. then refactor the now duplicated code: `ac08d0f53a83 <https://github.com/scality/metalk8s/commit/ac08d0f53a835a0b2bc61c1fe5b7317bf4d6550c>`_
+
+If you add tests (and you should), they can be introduced in a dedicated commit
+as well.
+
+Mixing unrelated changes
+::::::::::::::::::::::::
+
+It is sometimes tempting to do small unrelated changes as you are working on
+something else in the same code area.
+Please refrain to do so, or at least do it in a dedicated commit.
+
+Mixing non-related changes into the same commit makes revert and cherry-pick
```

Comment:
And understanding as well!

Reply:
I'll add it, good point :slightly_smiling_face: